api/v1/client/src/commonMain/kotlin/HttpClientUtils.kt

```diff
@@ -33,12 +33,14 @@ import io.ktor.serialization.kotlinx.json.json
 
 import kotlinx.serialization.json.Json
 
+internal expect fun getDefaultHttpClientEngine(): HttpClientEngine
+
 /**
  * Create a default HTTP client with the given [json] configuration and [engine]. If no engine is provided, it will
  * choose the engine automatically based on the platform.
  */
 fun createDefaultHttpClient(json: Json = Json.Default, engine: HttpClientEngine? = null): HttpClient {
-    val client = engine?.let { HttpClient(it) } ?: HttpClient()
+    val client = engine?.let { HttpClient(it) } ?: HttpClient(getDefaultHttpClientEngine())
```

Comment:
I don't think that this is needed. `HttpClient` will choose the correct engine at compile time based on the target platform (see [here](https://ktor.io/docs/client-engines.html#default)).

Reply:
Yes, you're right. I just need to have the dependency specified, then it selects the correct implementation.
Changed.